TechnicalDocuments/CompomentComposition.md

```diff
@@ -0,0 +1,64 @@
+
+The purpose of this document is to provide documentation for __composing Bento components using `UIStackView`__.
+Other ways of composition are out of the scope of this document.
+
+## How to use component composition in your framework
+
+Well, the answer to this question is that your can't.
+The reasoning for this decision is that
+* we must reuse components via the `DesignLibrary`
+* If you need something custom and customizable then using `UIStackView`s probably won't be good enough for you
+* We need to be flexible and to depend on abstractions.
+
+#### Where and how to use component composition
+`BabylonUI` now has a new namespace called `UI`, which contains a few `atomic` components.
+
+- `Atomic Components` are the `Bento` equivalent of standard `UIKit` components — this is why they retain a similar nomenclature (e.g: `UILabel` and `UI.Label`, `UIButton` and `UI.Button`);
+- `Atomic Components` use `Stylesheets` for styling;
+- `Atomic Components` should behave like their `UIKit` counterpart.
+
+## Code Sample
+
+```swift
+[
+    UI.Label(
+      text: "Some Title",
+      styleSheet: LabelStyleSheet()
+          .compose(\.textColor, tokens.colors.black)
+          .compose(\.font, font(fontAttributes: tokens.typography.body))
+    ).asAnyRenderable(),
+    UI.Label(
+        text: "Some Details",
+        styleSheet: LabelStyleSheet()
+            .compose(\.textColor, tokens.colors.grey700)
+            .compose(\.font, font(fontAttributes: tokens.typography.subhead))
+    ).asAnyRenderable()
+]
+.stack(
+    axis: .vertical,
+    spacing: 2
+)
+```
+
+Currently `Renderable` provides some util methods like `card(..)`, `stack(...)`, `width(...)` and `height(...)`.
+You can also add new convenience methods if you need them. 
+
+## `UIStackView` based component composition depends on Auto Layout
+
+When you compose `Bento` components using `UIStackView` __there is an implicit behaviour change on how you create your UI__.
+
+So let's start with how it works
+
+| Atomic Components using UIStackView | BentoKit | Vanilla Bento |
+|:-------:|:------:|:------:|
+| <ul><li> Step 1: Compose your components like in the code sample. This step means that you use Auto Layout. </li><li> Step 2: render your `Box` </li> <li>Step 3: `UITableView`/`UICollectionView` use Auto Layout in order to calculate the size of the cell.</li></ul> | Step 1: Implement your components using Auto Layout. </li> <li> Step2: Implement `HeightCustomizing` and calculate the size of the component. <li> Step 3: render your `Box` </li> <li>Step 4: `UITableView`/`UICollectionView` use Auto Layout in order to calculate the size of the cell __if the component doesn't comfort to `HeightCustomizing` othewise it uses the provided value__.</li></ul>  |  <li>Step 1: Implement your components *and* use Auto Layout. </li><li> Step 2: render your `Box` </li> <li>Step 3: `UITableView`/`UICollectionView` use Auto Layout in order to calculate the size of the cell.</li></ul>  |
```

Comment:
Table format is not very convenient here, there separate booklet lists would be easier to read. It also have almost the same content for each column so probably what really needs to be explained is only why `HeightCustomising` thing is/was needed in the first place and how it affects the component, i.e. that the calculated height will be always preferred to the height set by other means, like `Height` component from Design Library.

Reply:
We could even maybe have a runtime warning when `height` is applied on a component that implements `HeightCustomising` protocol, if that's even possible to check. cc @sergdort

Reply:
👀